File: Python/codeup_1098_설탕과자 뽑기.py
w, h = input().split()
w = int(w)
h = int(h)

arr = [[0]*h for i in range(w)]

n = int(input())

for i in range(0, n):
    l, d, x, y = input().split()
    l = int(l)
    d = int(d)
    x = int(x)
    y = int(y)

    for j in range(0, l):
        if d == 0:
            arr[x - 1][y - 1] = 1;
            y += 1
        else:
            arr[x - 1][y - 1] = 1;
            x += 1

for i in range(w):
    for j in range(h):
        print(arr[i][j], end = ' ')
    print()

''' 파이썬 리스트 초기화 
m=[]
for i in range(20) :
    m.append([])
    for j in range(20) :
        m[i].append(0)
'''

File: C++/codeup_1098_설탕과자 뽑기.cpp
#include<iostream>
using namespace std;

int main()
{
    int w, h;
    int n;
    int l, d, x, y;

    cin >> h >> w;

    int **arr = new int *[h];
    for(int i = 0; i < h; i++)
        arr[i] = new int[w];

    for(int i = 0; i < h; i++)
        for(int j = 0; j < w; j++)
            arr[i][j] = 0;

    cin >> n;

    for(int i = 0; i < n; i++)
    {
        cin >> l >> d >> x >> y;

        for(int j = 0; j < l; j++)
        {
            if(d == 0)
            {
                arr[x - 1][y - 1] = 1;
                y++;
            }
            else
            {
                arr[x - 1][y - 1] = 1;
                x++;
            }
        }
    }

for(int i = 0; i < h; i++)
{
    for(int j = 0; j < w; j++)
    {
        cout << arr[i][j] << " ";
    }

    cout << endl;
}
    return 0;
}

/*
가로세로
막대개수
막대길이 방향 좌표

5 5 
3
2 0 1 1
3 1 2 3
4 1 2 5
*/
